primitives/src/unchecked_extrinsic.rs

```diff
@@ -78,38 +93,50 @@ impl<Call, Extra: SignedExtension, ConvertTx, StorageDepositPerByte, TxFeePerGas
 	}
 }
 
-impl<Call, Extra: SignedExtension, ConvertTx, StorageDepositPerByte, TxFeePerGas> ExtrinsicMetadata
-	for AcalaUncheckedExtrinsic<Call, Extra, ConvertTx, StorageDepositPerByte, TxFeePerGas>
+impl<Call, Extra: SignedExtension, EnableEvmSignature, ConvertTx, StorageDepositPerByte, TxFeePerGas, PayerTx>
+	ExtrinsicMetadata
+	for AcalaUncheckedExtrinsic<Call, Extra, EnableEvmSignature, ConvertTx, StorageDepositPerByte, TxFeePerGas, PayerTx>
 {
 	const VERSION: u8 = UncheckedExtrinsic::<Address, Call, AcalaMultiSignature, Extra>::VERSION;
 	type SignedExtensions = Extra;
 }
 
-impl<Call, Extra: SignedExtension, ConvertTx, StorageDepositPerByte, TxFeePerGas> ExtrinsicCall
-	for AcalaUncheckedExtrinsic<Call, Extra, ConvertTx, StorageDepositPerByte, TxFeePerGas>
+impl<Call, Extra: SignedExtension, EnableEvmSignature, ConvertTx, StorageDepositPerByte, TxFeePerGas, PayerTx>
+	ExtrinsicCall
+	for AcalaUncheckedExtrinsic<Call, Extra, EnableEvmSignature, ConvertTx, StorageDepositPerByte, TxFeePerGas, PayerTx>
 {
 	fn call(&self) -> &Self::Call {
 		self.0.call()
 	}
 }
 
-impl<Call, Extra, ConvertTx, StorageDepositPerByte, TxFeePerGas, Lookup> Checkable<Lookup>
-	for AcalaUncheckedExtrinsic<Call, Extra, ConvertTx, StorageDepositPerByte, TxFeePerGas>
+impl<Call, Extra, EnableEvmSignature, ConvertTx, StorageDepositPerByte, TxFeePerGas, Lookup, PayerTx> Checkable<Lookup>
+	for AcalaUncheckedExtrinsic<Call, Extra, EnableEvmSignature, ConvertTx, StorageDepositPerByte, TxFeePerGas, PayerTx>
 where
 	Call: Encode + Member,
 	Extra: SignedExtension<AccountId = AccountId32>,
+	EnableEvmSignature: crate::SwitchFlag,
 	ConvertTx: Convert<(Call, Extra), Result<(EthereumTransactionMessage, Extra), InvalidTransaction>>,
+	PayerTx: Convert<(Call, Extra), Result<(), InvalidTransaction>>,
 	StorageDepositPerByte: Get<Balance>,
 	TxFeePerGas: Get<Balance>,
 	Lookup: traits::Lookup<Source = Address, Target = AccountId32>,
 {
 	type Checked = CheckedExtrinsic<AccountId32, Call, Extra>;
 
 	fn check(self, lookup: &Lookup) -> Result<Self::Checked, TransactionValidityError> {
+		let function = self.0.function.clone();
+		let signature = self.0.signature.clone();
+		if let Some((_, _, extra)) = signature {
+			PayerTx::convert((function.clone(), extra))?;
+		}
+
+		if !EnableEvmSignature::value() {
```

Comment:
I think it is ok to not have this check. We are going to enable EvmSignature for Karura & Acala soon enough.